bindu/auth/hydra/registration.py

```
@@ -44,15 +37,18 @@ def save_agent_credentials(
 
-    # Save to file
+    # ✅ SIMPLE + CORRECT WRITE
     with open(creds_file, "w") as f:
         json.dump(existing_creds, f, indent=2)
 
-    # Set restrictive permissions (owner read/write only)
-    creds_file.chmod(0o600)
+    # 🔐 FORCE PERMISSIONS (this is the only addition)
+    try:
+        os.chmod(creds_file, 0o600)
+    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)